src/pages/stats/[login].tsx

```diff
@@ -1,328 +1,55 @@
-import { RepositoryContributionsCard } from "@/components";
-import { useGitHubPullRequests } from "@/hooks";
-import { PullRequestContributionsByRepository } from "@/types/github";
-import { exportAsImage } from "@/utils";
+import { useState } from "react";
 import { useSession } from "next-auth/react";
 import { useRouter } from "next/router";
-import { useMemo, useState } from "react";
+import { useGitHubPullRequests, useFilteredRepositories } from "@/hooks";
 import CardSkeleton from "@/components/CardSkeleton";
-import { toast } from "react-toastify";
-
-const yearsRange = 4;
+import ReposFilters from "@/components/ReposFilters";
+import FormatStatsRender from "@/components/FormatStatsRender";
```

Comment:
All components should be imported from `"@components"` only, don't forget to add them to `components/index.ts`

Reply:
The new components are imported from `"@components"` but I didn't add them  to `components/index.ts`, still they are correctly imported. Could you please tell me why do we need to add them  to `components/index.ts`? Thanks

Reply:
Exporting components (everything, in general) through an `index.ts` allows for better organized imports when those components are used.

It also hides any potential complexity in case there's a nested structure inside components (not our case right now though).

The steps in your case would be:
1. Remove `export default` from the components and export directly the const
2. Add `export * from "/path"` on index.ts
3. Improve the import in [login].tsx

(see RepositoryContributionsCard.tsx)

It becomes from this:
```
import CardSkeleton from "@/components/CardSkeleton";
import ReposFilters from "@/components/ReposFilters";
import FormatStatsRender from "@/components/FormatStatsRender";
```
to this:
```
import { CardSkeleton, FormatStatsRender, ReposFilters } from "@/components";
```